builder/bundles.go

```diff
@@ -322,12 +328,26 @@ func queryRpmName(packageCmd []string, pkgName string, repo string) string {
 	)
 	queryStringRpm = append(queryStringRpm, pkgName)
 	outBuf, _ := helpers.RunCommandOutputEnv(queryStringRpm[0], queryStringRpm[1:], []string{"LC_ALL=en_US.UTF-8"})
-	if outBuf.String() != "" {
-		rpm := strings.Split(outBuf.String(), "\n")
-		_, file := filepath.Split(rpm[0])
-		return file
+
+	if outBuf.String() == "" {
+		return "", fmt.Errorf("rpm not found for pkg: %s", pkgName)
+	}
+
+	out := strings.Split(outBuf.String(), "\n")
+	pURL, err := url.Parse(out[0])
+	if err != nil {
+		return "", err
+	}
+
+	var rpmFullPath string
+	if pURL.Scheme == "file" { // obtain the full path without the url scheme
+		pURL.Scheme = ""
+		rpmFullPath = pURL.String()
+	} else { // obtain only the rpm name and append to repo cache dir
+		_, rpm := filepath.Split(out[0])
+		rpmFullPath = filepath.Join(repos[repo].cacheDir, rpm)
```

Comment:
Instead of passing repos to the function, we can use dnfDownloadDir directly as it is global. This works too.

Reply:
@ashleshaAtrey @jwakre I might even consider making `repos map[string]repoInfo` a global variable. But maybe in a different PR after @jwakre 's MCA PR is merged.